client/signup/config/flows.js

```diff
@@ -238,11 +239,72 @@ const Flows = {
 			flow = filterDesignTypeInFlow( flow );
 		}
 
-		return flow;
+		return Flows.getABTestFilteredFlow( flowName, flow );
 	},
 
 	getFlows() {
 		return flows;
+	},
+
+	/**
+	 * Preload AB Test variations after a certain step has been completed.
+	 *
+	 * This gives the option to set the AB variation as late as possible in the
+	 * signup flow.
+	 *
+	 * Currently only the `main` flow is whitelisted.
+	 *
+	 * @param {String} flowName The current flow
+	 * @param {String} stepName The step that is being completed right now
+	 */
+	preloadABTestVariationsForStep( flowName, stepName ) {
```

Comment:
It does make sense to do it in one place :) I updated the code to do that, now the `preloadABTestVariationsForStep` is being called from within `getFlow`, so everything is in one place. 

> One other thing, in case a user navigates back and forth through the steps several times you should check if the new step already exists before injecting it. 

Thanks for pointing that out. I added a check to make sure the step is not already added before adding it in the flow. The `Flows` "class" will undergo a bit of refactoring when I have the time to get to it, to make adding/removing/replacing steps more streamlined than it is now. 

> I think that would be simpler and a little more clear than what you're doing with .resumingFlow.

`resumingFlow` is not related to going back/forth between the steps themselves, but when you start signup, finish some steps, leave and then come back later (e.g. browser crash, page reload, etc.). We don't want to modify an already partly completed flow, or push the test for people that have started before the test went live. Having the check prevents that from happening.